### PR TITLE
Added a few convenience methods and JobDataMap argument

### PR DESCRIPTION
**Pull request by kroesctrl**

### CODE REVIEW

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java [hunk 1]
```diff
@@ -109,18 +110,28 @@ public JobDetail createJobDetail(final Class<? extends Job> jobClass) {
    *     OsgpScheduler#createJobDetail(Class)}.
    * @param cronExpression A CRON-expression.
    * @param timeZone A Timezone for the CRON-expression.
+   * @param jobDataMap A map containing data that can be used by job.
    * @return A {@link Trigger} instance.
    */
   public Trigger createJobTrigger(
```
Comment (harrymiddelburgetall) [inline osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java:116]: This method is only used locally. Can be private, I think

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java [hunk 2]
```diff
@@ -131,7 +142,8 @@ public Trigger createJobTrigger(
    * @return A {@link Trigger} instance.
    */
   public Trigger createJobTrigger(final JobDetail jobDetail, final String cronExpression) {
```
Comment (harrymiddelburgetall) [inline osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java:144]: Is this method made public only to be available for test?

**Review by harrymiddelburgetall — CHANGES_REQUESTED** [review]

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java [hunk 3]
```diff
@@ -109,18 +110,28 @@ public JobDetail createJobDetail(final Class<? extends Job> jobClass) {
...
         .forJob(jobDetail)
-        .withIdentity(jobDetail.getKey().getName() + "-Trigger")
+        .withIdentity(this.getTriggerIdentity(jobDetail))
+        .usingJobData(jobDataMap)
         .forJob(jobDetail)
         .withSchedule(CronScheduleBuilder.cronSchedule(cronExpression).inTimeZone(timeZone))
         .build();
   }
 
+  public TriggerKey getTriggerKey(final Class<? extends Job> jobClass) {
```
[the remark this answers is not in the document]
Comment (kroesctrl) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java:130]: We use it in SMHE

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java (hunk 1 above)
Comment (kroesctrl) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java:116]: No we use it from SMHE

File: osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java (hunk 2 above)
Comment (kroesctrl) [reply osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/application/scheduling/OsgpScheduler.java:144]: No we use it in SMHE

**Review by jboon — APPROVED** [review]

**Review by harrymiddelburgetall — APPROVED** [review]

**Review by kevinsmeets — APPROVED** [review]

**Pull request merged**